### ISSUE TITLE
Removing axes and data by index

### ISSUE THREAD

**Issue by thomasaarholt** [hyperspy/hyperspy#2121]
I'm simulating some data, and the simulation software always returns a fixed number of axes - in this case five navigation axes and two signal axes.

Sometimes I don't iterate over one of the axes, resulting in a signal shape of (e.g) `(200, 5, 1, 1, 20|256, 256)`. In these cases I wish to remove the axes with size 1, as it makes it easier to handle later on. 

I don't want to use s.inav[], because I don't necessarily know the correct indices of the axes to be removed in the signal. Is there a way to get the index?

I've tried using s.axes_manager.remove(), but it only removes the axis, not also the underlying data, i.e.:

```python
>>> s = hs.signals.Signal2D(np.random.random((2,3,1,5,6,8,8)))
>>> s.axes_manager.remove(s.axes_manager[2])
>>> s.plot()
ValueError: axes don't match array
```

**Comment by francisco-dlp** [comment]
You could try
```python
s.squeeze()
```
Probably it'll be good to add this method to the User Guide, so marking this as a bug.

**Comment by jlaehne** [comment]
Note that `squeeze()` also had a bug that only the first zero-dimensional dimension was removed - as the axes_manager index was updated in this operation, the iteration missed a directly following zero-dimensional axis. Fixed in #2676.
